src/test/java/com/netgrif/application/engine/petrinet/service/ProcessRoleServiceTest.java

```diff
@@ -0,0 +1,105 @@
+import com.netgrif.application.engine.petrinet.domain.roles.ProcessRole;
+import com.netgrif.application.engine.petrinet.domain.throwable.MissingPetriNetMetaDataException;
+import com.netgrif.application.engine.petrinet.service.interfaces.IPetriNetService;
+import com.netgrif.application.engine.petrinet.service.interfaces.IProcessRoleService;
+import com.netgrif.application.engine.startup.SuperCreator;
+import com.netgrif.application.engine.workflow.domain.eventoutcomes.petrinetoutcomes.ImportPetriNetEventOutcome;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.List;
+import java.util.Set;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@SpringBootTest
+@ActiveProfiles({"test"})
+@ExtendWith(SpringExtension.class)
+class ProcessRoleServiceTest {
+
+    private final static String ROLE_IMPORT_ID = "process_role";
+
+    @Autowired
+    private TestHelper testHelper;
+
+    @Autowired
+    private IPetriNetService petriNetService;
+
+    @Autowired
+    private IProcessRoleService processRoleService;
+
+    @Autowired
+    private SuperCreator superCreator;
+
+    @BeforeEach
+    public void before() {
+        testHelper.truncateDbs();
+    }
+
+
+    @Test
+    void shouldFindAllProcessRoles() throws IOException, MissingPetriNetMetaDataException {
+        petriNetService.importPetriNet(new FileInputStream("src/test/resources/all_data.xml"), VersionType.MAJOR, superCreator.getLoggedSuper());
+        petriNetService.importPetriNet(new FileInputStream("src/test/resources/role_all_data.xml"), VersionType.MAJOR, superCreator.getLoggedSuper());
+        List<ProcessRole> roles = processRoleService.findAll();
+        assertNotNull(roles);
+        assertFalse(roles.isEmpty());
+        assertEquals(6, roles.size());
+    }
+
+    @Test
+    void shouldFindAllProcessRolesByPetriNet() throws IOException, MissingPetriNetMetaDataException {
+        ImportPetriNetEventOutcome eventOutcome = petriNetService.importPetriNet(new FileInputStream("src/test/resources/all_data.xml"), VersionType.MAJOR, superCreator.getLoggedSuper());
+        List<ProcessRole> roles = processRoleService.findAll(eventOutcome.getNet().getStringId());
+        assertNotNull(roles);
+        assertFalse(roles.isEmpty());
+        assertEquals(1, roles.size());
+        assertEquals(ROLE_IMPORT_ID, roles.get(0).getImportId());
+    }
+
+    @Test
+    void shouldGetDefaultRole() {
+        ProcessRole role = processRoleService.defaultRole();
+        assertNotNull(role);
+        assertEquals(ProcessRole.DEFAULT_ROLE, role.getName().getDefaultValue());
```

Comment:
Add assert for `importId`. The default role is checked by name because historically the `importId` of the role was nowhere mentioned. I update it so it follows the same logic as the anonymous role but for backward compatibility, the method in `ProcessRoleService` has not been changed.